inc/Engine/License/API/Pricing.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace WP_Rocket\Engine\License\API;
+
+class Pricing {
+	/**
+	 * The pricing data object
+	 *
+	 * @var object
+	 */
+	private $pricing;
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param object $pricing The pricing object.
+	 */
+	public function __construct( $pricing ) {
+		$this->pricing = $pricing;
```

Comment:
Question: What is this object and where it coming from? 

If no from within Rocket, ignore the rest of this particular comment; else...

If the pricing object comes from within Rocket, change the design of the pricing object for us to go get the licensing type, etc. Why? This particular implementation is tightly coupled to know the deep details within the pricing object's data structure. By adding getters in the pricing object itself, the methods in this class (`Pricing)` do not need to know these details.